actors/runtime/src/main/java/cloud/orbit/actors/runtime/DefaultActorClassFinder.java

```diff
@@ -28,16 +28,73 @@
 
 package cloud.orbit.actors.runtime;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import cloud.orbit.actors.Actor;
+import cloud.orbit.actors.Remindable;
 import cloud.orbit.actors.extensions.ActorClassFinder;
+import io.github.lukehutch.fastclasspathscanner.FastClasspathScanner;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class DefaultActorClassFinder implements ActorClassFinder
 {
-    private static final ClassPathSearch search = new ClassPathSearch(Actor.class);
+    private static Logger logger = LoggerFactory.getLogger(DefaultActorClassFinder.class);
+
+    private static final Map<Class<?>, Class<?>> concreteImplementations = new ConcurrentHashMap<>();
+
+    static {
+        String basePackage = System.getProperty("orbit.actors.basePackage", "");
+        String[] scanSpec = new String[]{};
+        if (!basePackage.isEmpty())
+        {
+            String[] basePackages = basePackage.split(",");
+            Set<String> tmp = new HashSet<>(basePackages.length);
+            for (String s : basePackages)
+            {
+                if (!s.trim().isEmpty())
+                {
+                    tmp.add(s);
+                }
+            }
+            tmp.add("cloud.orbit"); // internal actors
+            scanSpec = tmp.toArray(new String[tmp.size()]);
```

Comment:
For efficiency, system, bootstrap and extension jarfiles like rt.jar (i.e. the jarfiles distributed with the JRE) are always blacklisted, as are package prefixes java and sun, i.e. they are never scanned. If you put custom classes into the lib/ext directory in your JRE folder (which is a valid but rare way of adding jarfiles to the classpath), they will be ignored by association with the JRE.